# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ShadowRoot.webidl
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 dom/webidl/ShadowRoot.webidl
--- a/dom/webidl/ShadowRoot.webidl	Wed May 15 03:01:18 2024 +0000
+++ b/dom/webidl/ShadowRoot.webidl	Wed May 15 03:01:19 2024 +0000
@@ -29,6 +29,8 @@
   readonly attribute SlotAssignmentMode slotAssignment;
   [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
   readonly attribute boolean clonable;
+  [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
+  readonly attribute boolean serializable;
   readonly attribute Element host;
   attribute EventHandler onslotchange;
 
@@ -62,6 +64,8 @@
   // https://html.spec.whatwg.org/#dom-shadowroot-sethtmlunsafe
   [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
   undefined setHTMLUnsafe(DOMString html);
+  [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
+  DOMString getHTML(optional GetHTMLOptions options = {});
 };
 
 ShadowRoot includes DocumentOrShadowRoot;